src/components/TimePicker.js

```diff
@@ -0,0 +1,420 @@
+};
+
+const AMOUNT_VIEW_ID = 'amountView';
+const NUM_PAD_CONTAINER_VIEW_ID = 'numPadContainerView';
+const NUM_PAD_VIEW_ID = 'numPadView';
+
+function formatHour(hourText) {
+    // // If the integer value of hour is greater than 12, return the second digit with a leading 0
+
+    // return twoDigitHour;
+    const hourNumber = parseInt(hourText, 10);
+
+    // If the integer value of hour is greater than 12, subtract 10
+    const adjustedHour = hourNumber > 12 ? hourNumber - 10 : hourNumber;
+
+    // Convert to a string and pad with a 0 if needed
+    return adjustedHour.toString().padStart(2, '0');
+}
+
+function insertAtPosition(originalString, newSubstring, selectionPositionFrom, selectionPositionTo, fn) {
+    // Check for invalid positions
+    if (selectionPositionFrom < 0 || selectionPositionTo < 0 || selectionPositionFrom > originalString.length || selectionPositionTo > originalString.length) {
+        return;
+    }
+
+    // If the positions are the same, it means we're inserting at a point
+    if (selectionPositionFrom === selectionPositionTo) {
+        if (selectionPositionFrom === originalString.length) {
+            if (typeof fn === 'function') fn();
+            return originalString; // If the insertion point is at the end, simply return the original string
+        }
+        return originalString.slice(0, selectionPositionFrom) + newSubstring + originalString.slice(selectionPositionFrom);
+    }
+
+    // Replace the selected range
+    return originalString.slice(0, selectionPositionFrom) + newSubstring + originalString.slice(selectionPositionTo);
+}
+
+function decreaseBothSelectionByOne({start, end}) {
+    if (start === 0) {
+        return {start: 0, end: 0};
+    }
+    return {start: start - 1, end: end - 1};
+}
+
+function replaceWithZeroAtPosition(originalString, position) {
+    if (position === 0 || position > 2) {
+        return originalString;
+    }
+    return `${originalString.slice(0, position - 1)}0${originalString.slice(position)}`;
+}
+
+function TimePicker({forwardedRef, value, errorText, onInputChange}) {
+    const {numberFormat} = useLocalize();
+    const [hours, setHours] = useState(DateUtils.parseTimeTo12HourFormat(value).hour);
+    const [minute, setMinute] = useState(DateUtils.parseTimeTo12HourFormat(value).minute);
+    const [selectionHour, setSelectionHour] = useState({start: 0, end: 0});
+    const [selectionMinute, setSelectionMinute] = useState({start: 2, end: 2}); // we focus it by default so need  to have selection on the end
+
+    const [amPmValue, setAmPmValue] = useState(DateUtils.getTimePeriod(value));
+
+    const hourInputRef = useRef(null);
+    const minuteInputRef = useRef(null);
+    const canUseTouchScreen = DeviceCapabilities.canUseTouchScreen();
+
+    const focusMinuteInputOnFirstCharacter = useCallback(() => {
+        minuteInputRef.current.focus();
+        setSelectionMinute({start: 0, end: 0});
+    }, []);
+
+    const handleHourChange = (text) => {
```

Comment:
This and `handleMinuteChange` feel a bit hard to parse at a glance. Perhaps some comments or tidy up is needed? It also seems like there's some code duplication between the two functions.